### ISSUE TITLE
Question: VEX-encoded SSE4 mentioned in `README.md`

### ISSUE THREAD

**Issue by AlexGuteniev** [google/highway#2029]
It says:

> If intending to use the AVX2 target together with half-width vectors (e.g. for `PromoteTo`), it is also important to compile with `/arch:AVX2`. This seems to be the only way to generate VEX-encoded SSE4 instructions on MSVC. Otherwise, mixing VEX-encoded AVX2 instructions and non-VEX SSE4 may cause severe performance degradation. Unfortunately, the resulting binary will then require AVX2. 

I haven't encountered a case where MSVC does not emit VEX-encoded SSE what it is followed by AVX (unless compiling with optimizations disabled).

My experiments has shown that it applies to SSE4 either, though i didn't test every instruction https://godbolt.org/z/9PP4onW34

Maybe you have a specific instruction combination in mind (which should be reported as a bug probably), or this information is obsolete?

**Comment by jan-wassenberg** [comment]
Hi, thanks for reaching out. Here is an example of the problem: https://godbolt.org/z/v8r54qzbq
Without the /arch:AVX2, we don't get the VEX encoding. Would you like to update the documentation, perhaps adding a link to this example?

**Comment by AlexGuteniev** [comment]
I don't think that your example illustrates the problem, as the overloads don't mix; they just coexist in the same TU, but not on the same code path.

If you try to mix them, either functions inline with encoding added, or `vzeroupper` is emitted https://godbolt.org/z/qvvsTd6en

**Comment by AlexGuteniev** [comment]
> Would you like to update the documentation, perhaps adding a link to this example?

Sure I can make a PR as soon as have understanding for myself.

**Comment by jan-wassenberg** [comment]
Here is what looks like a smoking gun, non-VEX + VEX in the same function, even though one was static+inlined: https://godbolt.org/z/shEfMEz5o

**Comment by AlexGuteniev** [comment]
Still not a problem, jsut a minor missing optimization, because non-VEX precede VEX, but do follow them.
If we repeat that after some VEX code, it will be `vpaddd`: https://godbolt.org/z/aMKea86ja

**Comment by AlexGuteniev** [comment]
If you're aware how to make MSVC generate code where non-VEX is followed by VEX without `vzeroupper`, then this is a serious bug, I would report it and link `README.md` to that. But I haven't encountered such situations so far.

**Comment by jan-wassenberg** [comment]
I agree it is a serious bug/shortcoming, and do not understand the lack of concern here. Once we have reached the smoking gun above, it is easy enough to call that right after VEX code. Also not too hard to get it into a single function: https://godbolt.org/z/EKP1of31q

**Comment by AlexGuteniev** [comment]
Thanks. I've learned the unfortunate truth and made a PR to describe it more precisely,
